app.py

```diff
@@ -39,75 +52,109 @@ def daily():
-	with open(CLIENT_FORMAT_FILE) as file_handle:
-		client_format = json.load(file_handle)
-	
-	# Specifies information about each sensor in the database
-	with open(DATABASE_FORMAT_FILE) as file_handle:
-		db_format = json.load(file_handle)
+	try:
+		tab = request.args.get('tab')
+		if not tab in client_format.keys(): raise ValueError()
+	except ValueError:
+		tab = next(iter(client_format))
 	
+	#print("==== " + date_str: tab + " ====" + "="*100)	
 	
-	# Get the data from the database corresponding to the current date
-	# >= and <= are used instead of == due to time zone issues
-	db_data = db.collection(DATABASE_COLLECTION).where("date", ">=", date).where("date", "<=", date+timedelta(1)).stream()
-	try:
-		readings = next(db_data).to_dict()["telemetry"] # The map within the read document
-		readings = OrderedDict(sorted(readings.items())) # Sort the map
-		print("readings: " + str(readings))
-	except StopIteration:
-		readings = {} # No document for day requested
-	
-	
-	graph_data = OrderedDict() # The data rearranged for usage in the render template (see format below)
-	""" graph_data {
-			"Battery": {
-				"Voltage": {"1579970374000": 423, "1579970379000": 419, ...},
-				"Current": {"1579970374000": 46, "1579970379000": 48, ...},
-				...
-			}, "Solar": ...
+	graph_data = OrderedDict() # The data used in the render template (see format below)
+	""" graph_data = {
+			"Voltage": {"1579970374000": 423, "1579970379000": 419, ...},
+			"Current": {"1579970374000": 46, "1579970379000": 48, ...},
 		} """
 	
+	# Loop through every sensor the current tab should show a reading for
+	for sensor_id in client_format[tab]["lines"]:
 	
-	# Tabs array represents each different graph tab the user can look at
-	tabs = []
-	for tab, tab_data in client_format.items():
-		tabs.append(tab)
-		graph_data[tab] = OrderedDict()
-		print("===== TAB: " + tab + " =====")
-		
-		# Loop through every sensor the current tab should show a reading for
-		for sensor_id in tab_data["lines"]:
-		
-			# Find the info about the sensor
-			sensor = next((item for item in db_format if item["id"] == sensor_id), None)
-			# Ensure the sensor is in the database
-			if sensor is not None and "index" in sensor:
-				graph_data[tab][sensor["name"]] = OrderedDict()
-				print("-- " + str(sensor["index"]) + ": " + sensor_id + " --")
-				
-				# Loop through all the sensor readings for the day being viewed
-				for time, reading in readings.items():					
-					unix = int(date.timestamp() + int(time))*1000 # TODO: Timezone incorrect
-					graph_data[tab][sensor["name"]][unix] = reading[sensor["index"]]
-					print(str(unix) + ": " + str(reading[sensor["index"]]))
+		# Find the info about the sensor
+		sensor = next((item for item in db_format if item["id"] == sensor_id), None)
+		# Ensure the sensor is in the database
+		if sensor is not None and "name" in sensor:
+			graph_data[sensor["name"]] = OrderedDict()
+			#print("-- " + str(sensor["index"]) + ": " + sensor_id + " --")
```

Comment:
Same here - if this needs to be left in the code then it's fine, but otherwise it should probably be removed.

Reply:
I'll go ahead and remove all the debug prints